test/standalone_test_arithmetic.hpp

```diff
@@ -0,0 +1,3206 @@
+///////////////////////////////////////////////////////////////
```

Comment:
Is it feasible to fold these changes into test_arithmetic.hpp (with some suitable pp-logic)?

I ask, because down the road it would be much too easy to modify one file and not the other.

Reply:
In the upcoming PR I am going to re-write this to use only the STL. There are still going to be confounding variables from using boost.test etc.

Reply:
There's no Boost.Test usage in the library, there's just a thin testing framework in test/test.hpp.  It does use boost's lightweight_test.hpp, but that should be trivial to emulate as there's not much in there.